lib/tlsca/ca.go

```diff
@@ -873,6 +883,19 @@ func (id *Identity) Subject() (pkix.Name, error) {
 		)
 	}
 
+	if id.JoinAttributes != nil {
+		encoded, err := protojson.Marshal(id.JoinAttributes)
+		if err != nil {
+			return pkix.Name{}, trace.Wrap(err, "encoding join attributes as protojson")
+		}
+		subject.ExtraNames = append(subject.ExtraNames,
+			pkix.AttributeTypeAndValue{
+				Type:  JoinAttributesASN1ExtensionOID,
+				Value: string(encoded),
```

Comment:
Can/Should we store this as bytes instead?

Reply:
I think the internal encoding implementation is identical (https://github.com/golang/go/blob/master/src/encoding/asn1/marshal.go#L41-L63) but I believe the values are explicitly typed. That doesn't matter much for us since we'll support whichever datatype we use, but I think tools (openssl, etc) can decide to render plaintext or hex depending on type. That is to say, I think strings are probably better due to the type that will be embedded alongside the value itself.

Technically I think ASN.1 has some native struct encoding we could use, and the go encoder supports it via field tags. That might be technically be "more" "correct" but probably adds too much complexity to be worth it.

Reply:
Yeah - I have a feeling that encoding as a string may just be the best bet. I think my biggest fear here is overly inflating certificate sizes - I'm yet to really determine the point at which it would become problematic. If testing does indicate this, then maybe we'll need to switch to encoding as binary rather than protojson.